root/components/ReleaseGroupsList.js

```diff
@@ -0,0 +1,263 @@
+        : '—'}
+    </td>
+    <td>
+      <EntityLink entity={releaseGroup} />
+    </td>
+    {showArtists ? (
+      <td>
+        <ArtistCreditLink
+          artistCredit={releaseGroup.artistCredit}
+        />
+      </td>
+    ) : null}
+    {noGrouping ? (
+      <td>
+        {releaseGroup.typeName
+          // TODO: replace with releaseGroupType when it works here
+          ? releaseGroup.l_type_name
+          : null
+        }
+      </td>
+    ) : null}
+    {noRatings ? null : (
+      <td className="c">
+        <RatingStars entity={releaseGroup} />
+      </td>
+    )}
+    <td className="c">{releaseGroup.release_count}</td>
+  </tr>
+);
+
+const ReleaseGroupsList = ({
+  $c,
+  checkboxes,
+  noGrouping,
+  noRatings,
+  order,
+  releaseGroups,
+  seriesItemNumbers,
+  showArtists,
+  sortable,
+}: ReleaseGroupsListProps) => {
+  const groupedReleaseGroups = groupBy(releaseGroups, 'typeName');
+  return (
+    noGrouping ? (
+      <table className="tbl release-group-list">
+        <ReleaseGroupsListHeader
+          $c={$c}
+          checkboxes={checkboxes}
+          noGrouping={noGrouping}
+          noRatings={noRatings}
+          order={order}
+          seriesItemNumbers={seriesItemNumbers}
+          showArtists={showArtists}
+          sortable={sortable}
+        />
+        <tbody>
+          {releaseGroups.map((releaseGroup, index) => (
+            <ReleaseGroupsListEntry
+              $c={$c}
+              checkboxes={checkboxes}
+              index={index}
+              key={releaseGroup.id}
+              noGrouping={noGrouping}
+              noRatings={noRatings}
+              releaseGroup={releaseGroup}
+              seriesItemNumbers={seriesItemNumbers}
+              showArtists={showArtists}
+            />
+          ))}
+        </tbody>
+      </table>
+    ) : (
+
+      Object.keys(groupedReleaseGroups).map((type) => {
+        const releaseGroupsOfType = groupedReleaseGroups[type];
+        return (
+          <React.Fragment key={type}>
+            <h3>
+              {type === 'null'
+                ? l('Unspecified type')
+                // TODO: replace with releaseGroupType when it works here
+                : releaseGroupsOfType[0].l_type_name
+              }
+            </h3>
+            <table className="tbl release-group-list">
+              <ReleaseGroupsListHeader
+                $c={$c}
+                checkboxes={checkboxes}
+                noGrouping={noGrouping}
+                noRatings={noRatings}
+                order={order}
+                seriesItemNumbers={seriesItemNumbers}
+                showArtists={showArtists}
+                sortable={sortable}
+              />
+              <tbody>
+                {releaseGroupsOfType.map((releaseGroup, index) => (
+                  <ReleaseGroupsListEntry
+                    $c={$c}
+                    checkboxes={checkboxes}
+                    index={index}
+                    key={releaseGroup.id}
```

Comment:
I'm not quite sure this key is required / useful.

Reply:
See https://reactjs.org/docs/lists-and-keys.html#keys